tracer/src/Datadog.Trace/Util/Http/HttpRequestExtensions.Framework.cs

```diff
@@ -27,17 +27,24 @@ internal static partial class HttpRequestExtensions
                 }
             }
 
-            var cookiesDic = new Dictionary<string, string>();
+            var cookiesDic = new Dictionary<string, string>(request.Cookies.AllKeys.Length);
             foreach (var k in request.Cookies.AllKeys)
             {
                 cookiesDic.Add(k, request.Cookies[k].Value);
             }
 
-            var dict = new Dictionary<string, object>()
+            var queryDic = new Dictionary<string, string>(request.QueryString.AllKeys.Length);
+            foreach (var k in request.QueryString.AllKeys)
+            {
+                var values = request.QueryString[k];
```

Comment:
no need to apply url decode on each element here?

Reply:
right, there's no need, it gets automatically applied when reading the query string this way.